### ISSUE TITLE
Duplicate products created instead of updating

### ISSUE THREAD

**Issue by lazyguru** [Agence-DnD/PIMGento#73]
It seems you are relying on your own table "pimgento_code" to decide if you need to create a product or not instead of using catalog_product_entity.  This is causing the code to try to create a new product for any product that was not initially created by PIMGento.  I was able to fix this by running the following SQL commands, but in my opinion you shouldn't use this table and should just check catalog_product_entity instead:

    delete from `pimgento_code`;
    truncate `pimgento_code`;
    insert into `pimgento_code` select null, 'product', `sku`, `entity_id` from `catalog_product_entity`;

**Comment by magentix** [comment]
Hi lazyguru,

Right, the product code (sku) is known by Magento.

In other imports (categories, options ...) we make the link between the code in Akeneo and the ID in Magento with "pimgento_code" table, in order not to change the table structure (for exemple: add column "code" in "eav_attribute_option").

So we kept the same logic for products, but products added in Magento will be duplicated instead of updating. We recommanded to delete all products in Magento before import from Akaneo, but your request is great if required.
